File: src/com/company/tree/MaximumDepth.java
package com.company.tree;

import java.util.ArrayList;
import java.util.List;

public class MaximumDepth {

    public static void main(String[] args) {
        List<Integer> list = new ArrayList<>();
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);

        int depth= maxDepth(root);

        System.out.println(depth);

    }

    private static int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int leftDepth = maxDepth(root.left);
        int rightDepth = maxDepth(root.right);

        return Math.max(leftDepth, rightDepth) + 1;

    }
}

File: src/com/company/tree/MaximumDepth.kt
package com.company.tree

fun maxDepth(root: TreeNode?): Int {
    root ?: return 0
    return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1
}

fun main(args: Array<String>) {
    val root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left?.left = TreeNode(4)
    root.left?.right = TreeNode(5)
    val depth = maxDepth(root)
    println(depth)
}
